aten/src/ATen/core/thread_pool.h

```diff
@@ -11,19 +15,74 @@ namespace ivalue {
 struct Future;
 } // namespace ivalue
 
-struct C10_API ThreadPool final {
-  ThreadPool();
+// TODO: move this to C10 and make it C10_API
+class CAFFE2_API TaskThreadPoolBase {
+ public:
+  virtual void run(const std::function<void()>& func) = 0;
+
+  virtual size_t size() const = 0;
+
```

Comment:
could you also add inThreadPool() function to this inteface, that checks that std::this_thread is one of the pool's threads